cfme/intelligence/optimization.py

```diff
@@ -0,0 +1,157 @@
+
+    @property
+    def is_displayed(self):
+        return (
+            self.logged_in_as_current_user
+            and self.title.text == "Optimization"
+            and self.table.is_displayed
+            and self.navigation.currently_selected == ["Overview", "Optimization"]
+            and self.breadcrumb.locations == ["Overview", "Optimization"]
+        )
+
+
+class OptimizationReportAllView(OptimizationView):
+    table = Table('//*[@id="main_div"]//table')
+
+    @property
+    def is_displayed(self):
+        menu_name = self.context["object"].menu_name
+        return (
+            self.logged_in_as_current_user
+            and self.title.text == menu_name
+            and self.navigation.currently_selected == ["Overview", "Optimization"]
+            and self.breadcrumb.locations == ["Overview", "Optimization", menu_name]
+        )
+
+
+class OptimizationSavedReportDetailsView(OptimizationView):
+    field = SelectorDropdown("id", "filterFieldTypeMenu")
+    search_text = SearchBox(locator='//input[contains(@placeholder, "search text")]')
+    table = Table('//*[@id="main_div"]//table')
+
+    @property
+    def is_displayed(self):
+        context_obj = self.context["object"]
+        return (
+            self.logged_in_as_current_user
+            and self.title.text == context_obj.name
+            and self.table.is_displayed
+            and self.navigation.currently_selected == ["Overview", "Optimization"]
+            and self.breadcrumb.locations
+            == ["Overview", "Optimization", context_obj.parent_obj.menu_name, context_obj.name]
+        )
+
+
+@attr.s
+class OptimizationReport(BaseEntity):
+    menu_name = attr.ib()
+    last_run_at = attr.ib(default=None)
+    runs = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        self._collections = {
+            "optimization_saved_reports": OptimizationSavedReportsCollection,
+            "saved_reports": SavedReportsCollection,
+            "reports": ReportsCollection,
+        }
+
+    def queue(self):
+        view = navigate_to(self.parent, "All")
+        row = view.table.row(report_name=self.menu_name)
+        row.action.widget.click()
+        view.flash.assert_no_error()
+        last_run_at = row["Last Run at"].text
+        wait_for(
+            lambda: row["Last Run at"].text != last_run_at,
+            fail_func=view.refresh.click,
+            num_sec=60,
+            message="Wait for report queue completion",
+            delay=1,
+        )
+        view.refresh.click()
+        self.last_run_at = row["Last Run at"].text
+        row.click()
+        view = self.create_view(OptimizationReportAllView)
+        return self.saved_optimization_reports.instantiate(
+            name=view.table.row(last_run_at=self.last_run_at)["Report"].text,
+            last_run_at=self.last_run_at,
+            username=self.appliance.user.credential.principal,
+        )
+
+    @cached_property
+    def saved_optimization_reports(self):
```

Comment:
What's the advantage to defining this cached property, when the reports collection can just be directly accessed via `self.collections` by whatever is calling this property?